### PR TITLE
Exclude redundant format setting error from coverage

### PR DESCRIPTION
**Pull request by awwad**
I probably don't need the check for an unsupported format (not 'json' or 'der') in common.sign_over_metadata; it's redundant at this depth in the code, and should be addressed within the called function in the TUF fork instead, but there's a tiny chance it'll spare someone some dev time.... The redundant check doesn't need to be hit in code coverage - even getting there with testing would require a very artificial procedure (set the format to something supported, then later switch it to something unsupported).

So I'm adding `#pragma no cover` to the error.

I'm doing the same for another spot in uptane.common, where a redundant check is performed on the key type of a key. Because the ANYKEY_SCHEMA is *defined* so that only supported keys can even be passed this far into the stack, it is neither currently possible to reach this point with an unsupported key type, nor likely to reach this point with an unsupported key type, nor practical to test reaching this point. Nonetheless, I'm leaving the condition in the code to guard against future changes.

### CODE REVIEW

File: uptane/common.py [hunk 1]
```diff
@@ -109,7 +109,7 @@ def sign_signable(
 
     # We should already be guaranteed to have a supported key type due to
     # the ANYKEY_SCHEMA.check_match call above. Defensive programming.
-    if signing_key['keytype'] not in SUPPORTED_KEY_TYPES:
+    if signing_key['keytype'] not in SUPPORTED_KEY_TYPES: # pragma no cover
```
Comment (vladimir-v-diaz) [inline uptane/common.py]: I think the correct comment to exlude code from coverage is `pragma: no cover`.  Please see coverage's documentation:
http://coverage.readthedocs.io/en/latest/excluding.html

File: uptane/common.py [hunk 2]
```diff
@@ -214,8 +214,10 @@ def sign_over_metadata(
   <Exceptions>
     tuf.FormatError, if 'key_dict' is improperly formatted.
 
-    tuf.UnsupportedLibraryError, if an unsupported or unavailable library is
-    detected.
+    tuf.UnsupportedLibraryError, if an unsupported or unavailable cryptography
+    library is chosen.
+
+    uptane.Error, if the given metadata format is not 'json' or 'der'
```
Comment (vladimir-v-diaz) [inline uptane/common.py:221]: You also raise `uptane.Error` if an unsupported keytype is specified in metadata.  See line 113.

File: uptane/common.py [hunk 3]
```diff
@@ -247,8 +249,8 @@ def sign_over_metadata(
         {'signed': data, 'signatures': []}, only_signed=True, datatype=datatype)
     data = hashlib.sha256(data).digest()
 
-  else:
-    raise tuf.Error('Unsupported metadata format: ' + repr(metadata_format))
+  else: # pragma no cover
```
Comment (vladimir-v-diaz) [inline uptane/common.py]: This should also be `# pragma: no cover`

File: uptane/common.py [hunk 4]
```diff
@@ -247,8 +249,8 @@ def sign_over_metadata(
...
+    raise uptane.Error('Unsupported metadata format: ' + repr(metadata_format))
```
Comment (vladimir-v-diaz) [inline uptane/common.py]: Consider printing the metadata formats that *are* supported.  The same also for unsupported key types (i.e., SUPPORTED_KEY_TYPES).

**Review by vladimir-v-diaz — CHANGES_REQUESTED** [review]

File: uptane/common.py (hunk 1 above)
Comment (awwad) [reply uptane/common.py]: Thanks for the catch.

File: uptane/common.py (hunk 2 above)
Comment (awwad) [reply uptane/common.py:221]: That's in a different function (and that function is not called by this one; only vice versa). I think I've added the errors to the appropriate places now.

**Pull request merged**